node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java

```diff
@@ -265,26 +273,32 @@ public Node allocate(ApplicationId owner, ClusterMembership membership, NodeReso
      */
     public Node with(Allocation allocation) {
         return new Node(id, ipConfig, hostname, parentHostname, flavor, status, state,
-                        Optional.of(allocation), history, type, reports, modelName);
+                        Optional.of(allocation), history, type, reports, modelName, reservedTo);
     }
 
     /** Returns a new Node without an allocation. */
     public Node withoutAllocation() {
         return new Node(id, ipConfig, hostname, parentHostname, flavor, status, state,
-                        Optional.empty(), history, type, reports, modelName);
+                        Optional.empty(), history, type, reports, modelName, reservedTo);
     }
 
 
     /** Returns a copy of this node with IP config set to the given value. */
     public Node with(IP.Config ipConfig) {
         return new Node(id, ipConfig, hostname, parentHostname, flavor, status, state,
-                        allocation, history, type, reports, modelName);
+                        allocation, history, type, reports, modelName, reservedTo);
     }
 
     /** Returns a copy of this node with the parent hostname assigned to the given value. */
     public Node withParentHostname(String parentHostname) {
         return new Node(id, ipConfig, hostname, Optional.of(parentHostname), flavor, status, state,
-                        allocation, history, type, reports, modelName);
+                        allocation, history, type, reports, modelName, reservedTo);
+    }
+
+    /** Returns a copy of this node marked as reserved to the given tenant (or empty to remove reservation) */
+    public Node withReservedTo(Optional<TenantName> reservedTo) {
```

Comment:
The convention is this class seems to be to have a `withReservedTo(TenantName)` and `withoutReservedTo()`

Reply:
Done